server/handlers/project.go

```diff
@@ -414,3 +414,63 @@ func ProjectDelete(userController controllers.UserController, projectController
 		ctx.JSON(http.StatusOK, gin.H{})
 	}
 }
+
+func ProjectLeave(userController controllers.UserController, projectController controllers.ProjectController, taskController controllers.TaskController, jwtParser *auth.JWTParser) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		id, _, ok := jwtParser.GetFromJWT(ctx)
+		if !ok {
+			DisplayNotAuthorized(ctx, "not logged in")
+			return
+		}
+		type Query struct {
+			Id  string `bson:"projectid" json:"projectid"`
+		}
+		var query Query
+		if err := ctx.BindJSON(&query); err != nil {
+			DisplayError(ctx, err.Error())
+			return
+		}
+		project, err := projectController.ProjectRetrieve(ctx, query.Id)
+		if err != nil {
+			DisplayError(ctx, err.Error())
+			return
+		}
+		user, err := userController.UserRetrieve(ctx, id, "")
+		if err == mongo.ErrNoDocuments {
+			DisplayError(ctx, "user does not exist")
+		} else if err != nil {
+			DisplayError(ctx, err.Error())
+		}
+		useridArr := []string{id}
+
+		// If user is admin, assign someone else admin role
+		if len(project.Members) > 1 {
+			if project.Settings.Roles[project.Members[id]].IsAdmin {
+				for nextUserId := range project.Members {
```

Comment:
works now because only 1 admin ever

but if we ever change that, this breaks

Reply:
ya I realised that while editing the docs.md. I was thinking this behavior is more relevant for a role like "owner". can have multiple admin, but only one owner